### PR TITLE
add uploader integration test

### PR DESCRIPTION
**Pull request by VinayGuthal**
Integration tests for the uploader

### CODE REVIEW

File: transport/transport-runtime/src/androidTest/java/com/google/android/datatransport/runtime/TestWorkScheduler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  private final Context context;
+
+  public TestWorkScheduler(Context applicationContext) {
+    this.context = applicationContext;
+  }
+
+  @Override
+  public void schedule(TransportContext transportContext, int attemptNumber) {
+    if (attemptNumber > 2) {
```
[the remark this answers is not in the document]
Comment (VinayGuthal) [reply transport/transport-runtime/src/androidTest/java/com/google/android/datatransport/runtime/TestWorkScheduler.java]: Talked offline and resolved

File: transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/AlarmManagerScheduler.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ public void schedule(TransportContext transportContext, int attemptNumber) {
     Long backendTime = eventStore.getNextCallTime(transportContext);
 
     long timeDiff = 0;
-    if (backendTime != null) {
+    if (backendTime != 0) {
```
Comment (vkryachko) [inline transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/AlarmManagerScheduler.java]: Is this check even needed? Would it make sense to replace with a single expression like: `long timeDiff = Math.max(0, backendTime - clock.getTime())`?

File: transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/JobInfoScheduler.java [hunk 3]
```diff
@@ -86,7 +86,7 @@ public void schedule(TransportContext transportContext, int attemptNumber) {
     Long backendTime = eventStore.getNextCallTime(transportContext);
 
     long timeDiff = 0;
-    if (backendTime != null) {
+    if (backendTime != 0) {
```
Comment (vkryachko) [inline transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/JobInfoScheduler.java]: Same here

File: transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/EventStore.java [hunk 4]
```diff
@@ -39,8 +39,7 @@
   void recordSuccess(Iterable<PersistedEvent> events);
 
   /** Returns the timestamp when the backend is allowed to be called next time or null. */
```
Comment (vkryachko) [inline transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/EventStore.java]: Please update the javadoc to indicate that we use zero for absense of value, not null.

**Pull request merged**